### PR TITLE
chore: add user agent

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

the app user agent does not include any useful information

### Solutions

add a customizable user agent

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: network/src/commonMain/kotlin/com/wire/kalium/network/networkContainer/AuthenticatedNetworkContainer.kt [hunk 1]
```diff
@@ -102,15 +102,19 @@ interface AuthenticatedNetworkContainer {
...
+            selfUserId: UserId,
+            userAgent: String,
         ): AuthenticatedNetworkContainer {
+
+            KaliumUserAgentProvider.setUserAgent(userAgent)
+
             return when (val version = sessionManager.serverConfig().metaData.commonApiVersion.version) {
                 0 -> AuthenticatedNetworkContainerV0(
-                    sessionManager
+                    sessionManager,
```
Comment (tmspzz) [inline network/src/commonMain/kotlin/com/wire/kalium/network/networkContainer/AuthenticatedNetworkContainer.kt:113]:
```suggestion
                    sessionManager
```

File: network/src/commonMain/kotlin/com/wire/kalium/network/networkContainer/AuthenticatedNetworkContainer.kt [hunk 2]
```diff
@@ -102,15 +102,19 @@ interface AuthenticatedNetworkContainer {
...
                 )
 
                 1 -> AuthenticatedNetworkContainerV0(
-                    sessionManager
+                    sessionManager,
```
Comment (tmspzz) [inline network/src/commonMain/kotlin/com/wire/kalium/network/networkContainer/AuthenticatedNetworkContainer.kt:117]:
```suggestion
                    sessionManager
```

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/v0/authenticated/networkContainer/AuthenticatedNetworkContainerV0.kt [hunk 3]
```diff
@@ -66,7 +66,7 @@ import io.ktor.client.engine.HttpClientEngine
 
 internal class AuthenticatedNetworkContainerV0 internal constructor(
     private val sessionManager: SessionManager,
-    engine: HttpClientEngine = defaultHttpEngine(sessionManager.serverConfig().links.apiProxy, sessionManager.proxyCredentials())
+    engine: HttpClientEngine = defaultHttpEngine(sessionManager.serverConfig().links.apiProxy, sessionManager.proxyCredentials()),
```
Comment (tmspzz) [inline network/src/commonMain/kotlin/com/wire/kalium/network/api/v0/authenticated/networkContainer/AuthenticatedNetworkContainerV0.kt:69]:
```suggestion
    engine: HttpClientEngine = defaultHttpEngine(sessionManager.serverConfig().links.apiProxy, sessionManager.proxyCredentials())
```

File: network/src/commonMain/kotlin/com/wire/kalium/network/networkContainer/UnboundNetworkContainer.kt [hunk 4]
```diff
@@ -37,16 +37,24 @@ private interface UnboundNetworkClientProvider {
...
     override val unboundNetworkClient by lazy {
         UnboundNetworkClient(engine)
     }
 }
 
 class UnboundNetworkContainerCommon(
-    val developmentApiEnabled: Boolean
-) : UnboundNetworkContainer, UnboundNetworkClientProvider by UnboundNetworkClientProviderImpl(developmentApiEnabled) {
+    val developmentApiEnabled: Boolean,
+    userAgent: String,
```
Comment (tmspzz) [inline network/src/commonMain/kotlin/com/wire/kalium/network/networkContainer/UnboundNetworkContainer.kt:55]:
```suggestion
    userAgent: String
```

**Review by tmspzz — CHANGES_REQUESTED** [review]

File: android/src/main/kotlin/com/wire/kalium/KaliumApplication.kt [hunk 5]
```diff
@@ -39,7 +39,8 @@ class KaliumApplication : Application(), Configuration.Provider {
         coreLogic = CoreLogic(
             appContext = applicationContext,
             rootPath = rootDir.absolutePath,
-            kaliumConfigs = KaliumConfigs()
+            kaliumConfigs = KaliumConfigs(),
+            userAgent = "Kalium test app"
```
Comment (alexandreferris) [inline android/src/main/kotlin/com/wire/kalium/KaliumApplication.kt:43]: Could we have the same as in darwin for example?
like: `userAgent = "Kalium Android"`

**Review by alexandreferris — APPROVED** [review]
Looking good 👌🏻 

Just have one question about the user agent name for android package

**Review by tmspzz — APPROVED** [review]

**Review by trOnk12 — APPROVED** [review]
impressive stuff <3

**Review by borichellow — APPROVED** [review]

**Pull request merged**